### PR TITLE
Add voucher value decision search filters

### PR DESCRIPTION
**Pull request by patari**
#### Summary

Same filters as in fee decision search.

### CODE REVIEW

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/data/VoucherValueDecisionQueries.kt [hunk 1]
```diff
@@ -305,23 +306,34 @@ fun Database.Read.searchValueDecisions(
...
             Binding.of("areas", areas),
             Binding.of("unit", unit),
+            Binding.of("postOffice", postOffice),
             Binding.of("start_date", startDate),
             Binding.of("end_date", endDate),
             Binding.of("financeDecisionHandlerId", financeDecisionHandlerId),
             Binding.of("difference", difference),
-            Binding.of("firstPlacementStartDate", evakaClock.now().toLocalDate().withDayOfMonth(1))
+            Binding.of("firstPlacementStartDate", evakaClock.now().toLocalDate().withDayOfMonth(1)),
+            if (retroactiveOnly) Binding.of("now", evakaClock.now()) else null
```
Comment (jaahon) [inline service/src/main/kotlin/fi/espoo/evaka/invoicing/data/VoucherValueDecisionQueries.kt:325]: This binding doesn't have to be included conditionally, there are other bindings already that are not always used in the query.

**Review by jaahon — COMMENTED** [review]
The way that the `UNCONFIRMED_HOURS` filter works now is a bit different compared to fee decisions. With fee decisions it's a property of the decision so the filter works the same way even if a child has their service need updates. In theory draft decisions should stay in sync with service need so the filter should work roughly the same for drafts but the behaviour is very different for sent and annulled decisions. I think both of the implementations should behave the same.

File: service/src/main/kotlin/fi/espoo/evaka/invoicing/data/VoucherValueDecisionQueries.kt (hunk 1 above)
Comment (patari) [reply service/src/main/kotlin/fi/espoo/evaka/invoicing/data/VoucherValueDecisionQueries.kt:325]: Yeah, should we add `service_need_missing` to `voucher_value_decision`-table?

**Comment by patari** [comment]
> The way that the `UNCONFIRMED_HOURS` filter works now is a bit different compared to fee decisions. With fee decisions it's a property of the decision so the filter works the same way even if a child has their service need updates. In theory draft decisions should stay in sync with service need so the filter should work roughly the same for drafts but the behaviour is very different for sent and annulled decisions. I think both of the implementations should behave the same.

Yeah, should we add `service_need_missing` to `voucher_value_decision`-table?

**Comment by jaahon** [comment]
> > The way that the `UNCONFIRMED_HOURS` filter works now is a bit different compared to fee decisions. With fee decisions it's a property of the decision so the filter works the same way even if a child has their service need updates. In theory draft decisions should stay in sync with service need so the filter should work roughly the same for drafts but the behaviour is very different for sent and annulled decisions. I think both of the implementations should behave the same.
> 
> Yeah, should we add `service_need_missing` to `voucher_value_decision`-table?

I think that would be the ideal solution.

**Review by jaahon — APPROVED** [review]

**Pull request merged**